x-pack/plugins/security_solution/server/lib/detection_engine/routes/rules/perform_bulk_action_route.test.ts

```diff
@@ -269,9 +291,9 @@ describe.each([
               ],
             },
           ],
+          results: getResultsStructure(),
```

Comment:
would be nice to actually check results content.

As with `getResultsStructure` looks like that:

```
  return {
    created: expect.any(Array),
    deleted: expect.any(Array),
    updated: expect.any(Array),
  };
```

we can't be fully confident in a test

Reply:
I don't think it would make it bloated.
What would mock for `results`in this particular test?
```
results: {
    created:[],
    deleted: [],
    updated: [],
}
```

is that correct? If it is, it doesn't look too much.

> And it should verify only this logic in isolation. We shouldn't try to cover all possible cases in one test suite.

Yes, we shouldn't, but the purpose of these tests(`perform_bulk_action_route.test.ts`) to test API's response as well and we omitting a chunk from it.

Reply:
I think a test that checks whether an error is returned when all index patterns are deleted should check only this logic. Checking something else in this test case doesn't add any value. We have a separate set of tests that check the rules update logic, including the `results` field.

We've already discussed tests a couple of times during our tech meetings. Everyone seemed to agree that tests should be as isolated as possible. But we can always reconsider our approach. So feel free to raise this topic during our next tech time.

Reply:
I don't think your example applies to this particular case.
Tests in this file tests body responses for particular error. Don't see how ensuring that results are correct, harm it. 

>  We have a separate set of tests that check the rules update logic, including the results field.

Are there any tests that test `results` in case of request failure? I think these ones are perfect for this. How else would you effectively do this? Functional tests? Aren't they too heavy for this?

Reply:
Ok, I'll try to clarify my position:

1. I'm not against increasing test coverage.
2. I'm firmly against mixing test cases and checking many irrelevant things in a single test case. When you check the entire response object, you're essentially creating a snapshot test with all the drawbacks snapshot tests have. They are brittle and produce a lot of false-positive failures on code changes (as you can already see in this PR, all test failures were false positive).
3. If the `results` field needs additional coverage, it should be done in a separate test case (no matter unit or integrational). So when this functionality changes, we have a single test failure instead of tens of randomly failing tests.

If you disagree with this approach and think we should leverage "snapshot"-style testing instead of focused tests, please raise this topic during our next tech time.

Reply:
As I mentioned before, now some chunk of response logic is missing from tests(results object)
I've suggested to test it within other tests as  I don't think it's irrelevant. If request fails partially, **we still need to return successfully updated items**. And in case of failure of all items, all arrays should be empty.
We also test whole responses in these tests, so it would make a perfect sense to add check of results there.

But if you prefer having a separate tests to cover results object, let's add them as separate test cases 👍

Reply:
> If request fails partially, we still need to return successfully updated items

> And in case of failure of all items, all arrays should be empty.

You just described two isolated test cases. They have nothing to do with `an error is returned if deleted all index patterns`. So let's test all of them separately.

Let's also address all outstanding test coverage issues in separate PRs.